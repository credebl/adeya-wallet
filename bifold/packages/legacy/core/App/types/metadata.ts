export enum CredentialMetadata {
  customMetadata = 'customMetadata',
}

export interface customMetadata {
  revoked_seen?: boolean
}
