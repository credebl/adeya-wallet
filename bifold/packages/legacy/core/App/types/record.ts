import { IndyRevocationInterval, CredentialExchangeRecord } from '@aries-framework/core'

export interface FieldParams {
  name: string | null
  format?: string
  type?: string
  encoding?: string
  mimeType?: string
  revoked?: boolean
  credentialId?: string
  label?: string
  restrictions?: Array<Record<string, unknown>>
  nonRevoked?: IndyRevocationInterval
}

export interface AttributeParams extends FieldParams {
  value: string | number | null
  revealed?: boolean
}

export interface PredicateParams extends FieldParams {
  pValue: string | number | null
  pType: string
  parameterizable?: boolean
}

export class Field {
  public name: string | null
  public format?: string
  public type?: string
  public encoding?: string
  public mimeType?: string
  public revoked?: boolean
  public credentialId?: string
  public label?: string
  public restrictions?: Array<Record<string, unknown>>
  public nonRevoked?: IndyRevocationInterval

  protected constructor(params: FieldParams) {
    this.name = params.name
    this.format = params.format
    this.type = params.type
    this.encoding = params.encoding
    this.mimeType = params.mimeType
    this.revoked = params.revoked
    this.credentialId = params.credentialId
    this.label = params.label
    this.restrictions = params.restrictions
    this.nonRevoked = params.nonRevoked
  }
}

export class Attribute extends Field {
  public value: string | number | null
  public revealed?: boolean

  public constructor(params: AttributeParams) {
    super(params)
    this.value = params.value
    this.revealed = params.revealed
  }
}

export class Predicate extends Field {
  public pValue: string | number | null
  public pType: string
  public parameterizable?: boolean

  public constructor(params: PredicateParams) {
    super(params)
    this.pValue = params.pValue
    this.pType = params.pType
    this.parameterizable = params.parameterizable
  }
}

export interface ProofCredentialAttributes {
  credExchangeRecord?: CredentialExchangeRecord
  credDefId?: string
  schemaId?: string
  credName: string
  attributes?: Attribute[]
}

export interface ProofCredentialPredicates {
  credExchangeRecord?: CredentialExchangeRecord
  credDefId?: string
  schemaId?: string
  credName: string
  predicates?: Predicate[]
}

export interface ProofCredentialItems extends ProofCredentialAttributes, ProofCredentialPredicates {}
