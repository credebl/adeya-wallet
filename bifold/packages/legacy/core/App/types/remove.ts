export enum ModalUsage {
  CredentialRemove = 1,
  ContactRemove,
  CredentialOfferDecline,
  ProofRequestDecline,
  CustomNotificationDecline,
}
