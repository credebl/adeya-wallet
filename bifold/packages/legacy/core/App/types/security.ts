export interface WalletSecret {
  id: string
  key?: string
  salt: string
}

export enum AuthLevel {
  BiometricsFallbackPIN = 'BiometricsFallbackPIN',
  BiometricsAndPIN = 'BiometricsAndPIN',
  BiometricsOnly = 'BiometricsOnly',
}

export interface PINValidationRules {
  only_numbers: boolean
  min_length: number
  max_length: number
  no_repeated_numbers: boolean | number
  no_repetition_of_the_two_same_numbers: boolean | number
  no_series_of_numbers: boolean
  no_even_or_odd_series_of_numbers: boolean
  no_cross_pattern: boolean
}

export interface PINSecurityParams {
  rules: PINValidationRules
  displayHelper: boolean
}
