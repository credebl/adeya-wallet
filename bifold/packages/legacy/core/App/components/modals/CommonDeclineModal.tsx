import * as React from 'react'
import { useTranslation } from 'react-i18next'
import { Modal, Platform, ScrollView, StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import { SafeAreaView } from 'react-native-safe-area-context'
import Icon from 'react-native-vector-icons/MaterialIcons'

import CredentialDeclined from '../../assets/img/credential-declined.svg'
import CustomNotificationDecline from '../../assets/img/delete-notification.svg'
import ProofDeclined from '../../assets/img/proof-declined.svg'
import { useTheme } from '../../contexts/theme'
import { GenericFn } from '../../types/fn'
import { ModalUsage } from '../../types/remove'
import { testIdWithKey } from '../../utils/testable'
import Button, { ButtonType } from '../buttons/Button'

interface CommonDeclineModalProps {
  usage: ModalUsage
  onSubmit?: GenericFn
  onCancel?: GenericFn
  visible?: boolean
}

const CommonDeclineModal: React.FC<CommonDeclineModalProps> = ({ usage, visible, onSubmit, onCancel }) => {
  if (!usage) {
    throw new Error('usage cannot be undefined')
  }

  const { t } = useTranslation()
  const { ColorPallet, TextTheme } = useTheme()

  const imageDisplayOptions = {
    height: 115,
    width: 115,
  }
  const styles = StyleSheet.create({
    container: {
      height: '100%',
      backgroundColor: ColorPallet.brand.modalPrimaryBackground,
      padding: 20,
    },
    controlsContainer: {
      marginTop: 'auto',
      marginHorizontal: 20,
      marginBottom: Platform.OS === 'ios' ? 108 : 20,
    },
    headerView: {
      alignItems: 'flex-end',
      marginTop: 65,
      backgroundColor: ColorPallet.brand.modalPrimaryBackground,
      height: 55,
      paddingTop: 10,
      paddingRight: 20,
      borderTopRightRadius: 10,
      borderTopLeftRadius: 10,
    },
    bodyText: {
      ...TextTheme.modalNormal,
      fontSize: 24,
      marginTop: 25,
    },
  })

  const headerImageForType = (usageType: ModalUsage) => {
    return (
      <View style={{ flexDirection: 'row', justifyContent: 'center' }}>
        {usageType === ModalUsage.CredentialOfferDecline && <ProofDeclined {...imageDisplayOptions} />}
        {usageType === ModalUsage.ProofRequestDecline && <CredentialDeclined {...imageDisplayOptions} />}
        {usageType === ModalUsage.CustomNotificationDecline && (
          <CustomNotificationDecline style={{ marginBottom: 15 }} {...imageDisplayOptions} />
        )}
      </View>
    )
  }

  const contentForType = (usageType: ModalUsage) => {
    if (usageType === ModalUsage.CredentialOfferDecline) {
      return (
        <View style={[{ marginBottom: 25 }]}>
          <Text style={[TextTheme.modalTitle, { fontSize: 28 }]}>{t('CredentialOffer.DeclineTitle')}</Text>
          <Text style={[styles.bodyText, { marginTop: 30 }]}>{t('CredentialOffer.DeclineParagraph1')}</Text>
          <Text style={[styles.bodyText]}>{t('CredentialOffer.DeclineParagraph1')}</Text>
        </View>
      )
    }

    if (usageType === ModalUsage.ProofRequestDecline) {
      return (
        <View style={[{ marginBottom: 25 }]}>
          <Text style={[TextTheme.modalTitle, { fontSize: 28 }]}>{t('ProofRequest.DeclineTitle')}</Text>
          <Text style={[styles.bodyText, { marginTop: 30 }]}>{t('ProofRequest.DeclineBulletPoint1')}</Text>
          <Text style={[styles.bodyText]}>{t('ProofRequest.DeclineBulletPoint2')}</Text>
          <Text style={[styles.bodyText]}>{t('ProofRequest.DeclineBulletPoint3')}</Text>
        </View>
      )
    }

    if (usageType === ModalUsage.CustomNotificationDecline) {
      return (
        <View style={[{ marginBottom: 25 }]}>
          <Text style={[TextTheme.modalTitle, { fontSize: 28 }]}>{t('CredentialOffer.CustomOfferTitle')}</Text>
          <Text style={[styles.bodyText, { marginTop: 30 }]}>{t('CredentialOffer.CustomOfferParagraph1')}</Text>
          <Text style={[styles.bodyText]}>{t('CredentialOffer.CustomOfferParagraph2')}</Text>
        </View>
      )
    }
  }

  return (
    <Modal transparent={true} visible={visible} animationType="slide">
      <View style={[styles.headerView]}>
        <TouchableOpacity accessibilityLabel="Close" testID="Close" onPress={() => onCancel && onCancel()}>
          <Icon name={'close'} size={42} color={TextTheme.modalNormal.color} />
        </TouchableOpacity>
      </View>
      <SafeAreaView
        edges={['left', 'right', 'bottom']}
        style={[
          {
            backgroundColor: ColorPallet.brand.modalPrimaryBackground,
          },
        ]}
      >
        <ScrollView style={[styles.container]}>
          {headerImageForType(usage)}
          {contentForType(usage)}
        </ScrollView>
        <View style={[styles.controlsContainer]}>
          <View style={[{ paddingTop: 10 }]}>
            <Button
              title={t('Global.Decline')}
              accessibilityLabel={t('Global.Decline')}
              testID={testIdWithKey('ConfirmDeclineButton')}
              onPress={() => onSubmit && onSubmit()}
              buttonType={ButtonType.ModalCritical}
            />
          </View>
          <View style={[{ paddingTop: 10 }]}>
            <Button
              title={t('Global.Cancel')}
              accessibilityLabel={t('Global.Cancel')}
              testID={testIdWithKey('CancelDeclineButton')}
              onPress={() => onCancel && onCancel()}
              buttonType={ButtonType.ModalSecondary}
            />
          </View>
        </View>
      </SafeAreaView>
    </Modal>
  )
}

export default CommonDeclineModal
