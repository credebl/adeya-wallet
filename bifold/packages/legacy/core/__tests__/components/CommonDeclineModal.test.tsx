import { render, fireEvent } from '@testing-library/react-native'
import React from 'react'

import CommonDeclineModal from '../../App/components/modals/CommonDeclineModal'
import { ModalUsage } from '../../App/types/remove'
import { testIdWithKey } from '../../App/utils/testable'

describe('CommonDeclineModal Component', () => {
  test('Rerenders correctly when not visible', async () => {
    const tree = render(<CommonDeclineModal visible={true} usage={ModalUsage.CredentialOfferDecline} />)

    expect(tree).toMatchSnapshot()
  })

  test('Controls trigger callbacks', async () => {
    const onSubmit = jest.fn()
    const onCancel = jest.fn()
    const tree = render(
      <CommonDeclineModal
        onSubmit={onSubmit}
        onCancel={onCancel}
        visible={true}
        usage={ModalUsage.CredentialOfferDecline}
      />
    )

    const confirmButton = tree.getByTestId(testIdWithKey('ConfirmDeclineButton'))
    const cancelDeclineButton = tree.getByTestId(testIdWithKey('CancelDeclineButton'))

    fireEvent(confirmButton, 'press')
    fireEvent(cancelDeclineButton, 'press')

    expect(onSubmit).toBeCalledTimes(1)
    expect(onCancel).toBeCalledTimes(1)
  })

  test('Credential offer decline renders correctly', async () => {
    const tree = render(<CommonDeclineModal visible={true} usage={ModalUsage.CredentialOfferDecline} />)

    expect(tree).toMatchSnapshot()
  })

  test('Proof request decline renders correctly', async () => {
    const tree = render(<CommonDeclineModal visible={true} usage={ModalUsage.ProofRequestDecline} />)

    expect(tree).toMatchSnapshot()
  })

  test('Custom notification decline renders correctly', async () => {
    const tree = render(<CommonDeclineModal visible={true} usage={ModalUsage.CustomNotificationDecline} />)

    expect(tree).toMatchSnapshot()
  })
})
