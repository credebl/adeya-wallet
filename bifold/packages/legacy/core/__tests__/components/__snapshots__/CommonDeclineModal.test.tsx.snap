// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonDeclineModal Component Credential offer decline renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-end",
          "backgroundColor": "#000000",
          "borderTopLeftRadius": 10,
          "borderTopRightRadius": 10,
          "height": 55,
          "marginTop": 65,
          "paddingRight": 20,
          "paddingTop": 10,
        },
      ]
    }
  >
    <View
      accessibilityLabel="Close"
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="Close"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 42,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <RNCSafeAreaView
    edges={
      Array [
        "left",
        "right",
        "bottom",
      ]
    }
    style={
      Array [
        Object {
          "backgroundColor": "#000000",
        },
      ]
    }
  >
    <RCTScrollView
      style={
        Array [
          Object {
            "backgroundColor": "#000000",
            "height": "100%",
            "padding": 20,
          },
        ]
      }
    >
      <View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <
            height={115}
            width={115}
          />
        </View>
        <View
          style={
            Array [
              Object {
                "marginBottom": 25,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 24,
                  "fontWeight": "bold",
                },
                Object {
                  "fontSize": 28,
                },
              ]
            }
          >
            CredentialOffer.DeclineTitle
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 24,
                  "fontWeight": "normal",
                  "marginTop": 25,
                },
                Object {
                  "marginTop": 30,
                },
              ]
            }
          >
            CredentialOffer.DeclineParagraph1
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 24,
                  "fontWeight": "normal",
                  "marginTop": 25,
                },
              ]
            }
          >
            CredentialOffer.DeclineParagraph1
          </Text>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Array [
          Object {
            "marginBottom": 108,
            "marginHorizontal": 20,
            "marginTop": "auto",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "paddingTop": 10,
            },
          ]
        }
      >
        <View
          accessibilityLabel="Global.Decline"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#42803E",
              "borderRadius": 4,
              "opacity": 1,
              "padding": 16,
            }
          }
          testID="com.ariesbifold:id/ConfirmDeclineButton"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              Global.Decline
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "paddingTop": 10,
            },
          ]
        }
      >
        <View
          accessibilityLabel="Global.Cancel"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderColor": "#42803E",
              "borderRadius": 4,
              "borderWidth": 2,
              "opacity": 1,
              "padding": 16,
            }
          }
          testID="com.ariesbifold:id/CancelDeclineButton"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#42803E",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              Global.Cancel
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</Modal>
`;

exports[`CommonDeclineModal Component Custom notification decline renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-end",
          "backgroundColor": "#000000",
          "borderTopLeftRadius": 10,
          "borderTopRightRadius": 10,
          "height": 55,
          "marginTop": 65,
          "paddingRight": 20,
          "paddingTop": 10,
        },
      ]
    }
  >
    <View
      accessibilityLabel="Close"
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="Close"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 42,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <RNCSafeAreaView
    edges={
      Array [
        "left",
        "right",
        "bottom",
      ]
    }
    style={
      Array [
        Object {
          "backgroundColor": "#000000",
        },
      ]
    }
  >
    <RCTScrollView
      style={
        Array [
          Object {
            "backgroundColor": "#000000",
            "height": "100%",
            "padding": 20,
          },
        ]
      }
    >
      <View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <
            height={115}
            style={
              Object {
                "marginBottom": 15,
              }
            }
            width={115}
          />
        </View>
        <View
          style={
            Array [
              Object {
                "marginBottom": 25,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 24,
                  "fontWeight": "bold",
                },
                Object {
                  "fontSize": 28,
                },
              ]
            }
          >
            CredentialOffer.CustomOfferTitle
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 24,
                  "fontWeight": "normal",
                  "marginTop": 25,
                },
                Object {
                  "marginTop": 30,
                },
              ]
            }
          >
            CredentialOffer.CustomOfferParagraph1
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 24,
                  "fontWeight": "normal",
                  "marginTop": 25,
                },
              ]
            }
          >
            CredentialOffer.CustomOfferParagraph2
          </Text>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Array [
          Object {
            "marginBottom": 108,
            "marginHorizontal": 20,
            "marginTop": "auto",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "paddingTop": 10,
            },
          ]
        }
      >
        <View
          accessibilityLabel="Global.Decline"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#42803E",
              "borderRadius": 4,
              "opacity": 1,
              "padding": 16,
            }
          }
          testID="com.ariesbifold:id/ConfirmDeclineButton"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              Global.Decline
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "paddingTop": 10,
            },
          ]
        }
      >
        <View
          accessibilityLabel="Global.Cancel"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderColor": "#42803E",
              "borderRadius": 4,
              "borderWidth": 2,
              "opacity": 1,
              "padding": 16,
            }
          }
          testID="com.ariesbifold:id/CancelDeclineButton"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#42803E",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              Global.Cancel
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</Modal>
`;

exports[`CommonDeclineModal Component Proof request decline renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-end",
          "backgroundColor": "#000000",
          "borderTopLeftRadius": 10,
          "borderTopRightRadius": 10,
          "height": 55,
          "marginTop": 65,
          "paddingRight": 20,
          "paddingTop": 10,
        },
      ]
    }
  >
    <View
      accessibilityLabel="Close"
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="Close"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 42,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <RNCSafeAreaView
    edges={
      Array [
        "left",
        "right",
        "bottom",
      ]
    }
    style={
      Array [
        Object {
          "backgroundColor": "#000000",
        },
      ]
    }
  >
    <RCTScrollView
      style={
        Array [
          Object {
            "backgroundColor": "#000000",
            "height": "100%",
            "padding": 20,
          },
        ]
      }
    >
      <View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <
            height={115}
            width={115}
          />
        </View>
        <View
          style={
            Array [
              Object {
                "marginBottom": 25,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 24,
                  "fontWeight": "bold",
                },
                Object {
                  "fontSize": 28,
                },
              ]
            }
          >
            ProofRequest.DeclineTitle
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 24,
                  "fontWeight": "normal",
                  "marginTop": 25,
                },
                Object {
                  "marginTop": 30,
                },
              ]
            }
          >
            ProofRequest.DeclineBulletPoint1
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 24,
                  "fontWeight": "normal",
                  "marginTop": 25,
                },
              ]
            }
          >
            ProofRequest.DeclineBulletPoint2
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 24,
                  "fontWeight": "normal",
                  "marginTop": 25,
                },
              ]
            }
          >
            ProofRequest.DeclineBulletPoint3
          </Text>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Array [
          Object {
            "marginBottom": 108,
            "marginHorizontal": 20,
            "marginTop": "auto",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "paddingTop": 10,
            },
          ]
        }
      >
        <View
          accessibilityLabel="Global.Decline"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#42803E",
              "borderRadius": 4,
              "opacity": 1,
              "padding": 16,
            }
          }
          testID="com.ariesbifold:id/ConfirmDeclineButton"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              Global.Decline
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "paddingTop": 10,
            },
          ]
        }
      >
        <View
          accessibilityLabel="Global.Cancel"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderColor": "#42803E",
              "borderRadius": 4,
              "borderWidth": 2,
              "opacity": 1,
              "padding": 16,
            }
          }
          testID="com.ariesbifold:id/CancelDeclineButton"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#42803E",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              Global.Cancel
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</Modal>
`;

exports[`CommonDeclineModal Component Rerenders correctly when not visible 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-end",
          "backgroundColor": "#000000",
          "borderTopLeftRadius": 10,
          "borderTopRightRadius": 10,
          "height": 55,
          "marginTop": 65,
          "paddingRight": 20,
          "paddingTop": 10,
        },
      ]
    }
  >
    <View
      accessibilityLabel="Close"
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="Close"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 42,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <RNCSafeAreaView
    edges={
      Array [
        "left",
        "right",
        "bottom",
      ]
    }
    style={
      Array [
        Object {
          "backgroundColor": "#000000",
        },
      ]
    }
  >
    <RCTScrollView
      style={
        Array [
          Object {
            "backgroundColor": "#000000",
            "height": "100%",
            "padding": 20,
          },
        ]
      }
    >
      <View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <
            height={115}
            width={115}
          />
        </View>
        <View
          style={
            Array [
              Object {
                "marginBottom": 25,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 24,
                  "fontWeight": "bold",
                },
                Object {
                  "fontSize": 28,
                },
              ]
            }
          >
            CredentialOffer.DeclineTitle
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 24,
                  "fontWeight": "normal",
                  "marginTop": 25,
                },
                Object {
                  "marginTop": 30,
                },
              ]
            }
          >
            CredentialOffer.DeclineParagraph1
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 24,
                  "fontWeight": "normal",
                  "marginTop": 25,
                },
              ]
            }
          >
            CredentialOffer.DeclineParagraph1
          </Text>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Array [
          Object {
            "marginBottom": 108,
            "marginHorizontal": 20,
            "marginTop": "auto",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "paddingTop": 10,
            },
          ]
        }
      >
        <View
          accessibilityLabel="Global.Decline"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#42803E",
              "borderRadius": 4,
              "opacity": 1,
              "padding": 16,
            }
          }
          testID="com.ariesbifold:id/ConfirmDeclineButton"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              Global.Decline
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "paddingTop": 10,
            },
          ]
        }
      >
        <View
          accessibilityLabel="Global.Cancel"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderColor": "#42803E",
              "borderRadius": 4,
              "borderWidth": 2,
              "opacity": 1,
              "padding": 16,
            }
          }
          testID="com.ariesbifold:id/CancelDeclineButton"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#42803E",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              Global.Cancel
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</Modal>
`;
